python/lsst/ip/isr/isrTask.py

```diff
@@ -661,10 +661,16 @@ def updateVariance(self, ampExposure, amp):
         @param[in,out]  ampExposure     exposure to process
         @param[in]      amp             amplifier detector information
         """
-        if not math.isnan(amp.getGain()):
+        gain = amp.getGain()
+        if not math.isnan(gain):
+            if gain <= 0:
```

Comment:
Why not combine these lines as `if not math.isnan(gain) and gain <= 0:`?

Reply:
Because the action on NaN and <= 0 is different.

Reply:
is there anything further to do on this?